src/repo_smith/helpers/files_helper.py

```diff
@@ -0,0 +1,50 @@
+import os
+import shutil
+import stat
+import textwrap
+from io import TextIOWrapper
+from typing import Optional
+
+from git import Repo
+
+from repo_smith.helpers.helper import Helper
+from repo_smith.types import FilePath
+
+
+class FilesHelper(Helper):
+    def __init__(self, repo: Repo, verbose: bool) -> None:
+        super().__init__(repo, verbose)
+
+    def create_or_update(
+        self, filepath: FilePath, contents: Optional[str] = None
+    ) -> None:
+        """Creates or updates a file with the given content."""
+        dirname = os.path.dirname(filepath)
+        if dirname != "":
+            os.makedirs(dirname, exist_ok=True)
+
+        if contents is None:
+            open(filepath, "a").close()
```

Comment:
```suggestion
            with open(filepath, "a"):
                pass
```

Should we use `with` pattern here instead as it is considered safer, as this can lead to errors if an exception occurs between open and close file.

Reply:
The `with` in this case doesn't add anything. It would simply call on `open()` and `.close()` on our behalf. I personally prefer not expanding a block if possible since we don't need the scope that the `with` gives us

Reply:
That makes sense, in any case it works. The edge case I was thinking was

`open()` -> exception occurs here -> means file is never opened so it should be fine, there is no need for `close()`